pages/add-user.page.ts

```diff
@@ -0,0 +1,86 @@
+import { Locator, Page } from "@playwright/test";
+import { extractSelectedDisplayedValue } from "../helpers/extractSelectedDisplayedValue";
+import { URLS } from "../config/urlProvider";
+
+export class AddUserPage {
+  readonly page: Page;
+  readonly createBtn: Locator;
+  readonly cancelBtn: Locator;
+  readonly userNameField: Locator;
+  readonly yearOfBirthField: Locator;
+  readonly genderField: Locator;
+  readonly headerListitem: Locator;
+  readonly maxUserNameLength: number;
+  readonly minUserNameLength: number;
+  public yearOfBirthFieldError: Locator;
+  public userNameFieldError: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.createBtn = this.page.locator(
+      'xpath=//button[@data-testid="button-Create"]',
+    );
+    this.cancelBtn = this.page.locator(
+      'xpath=//a[@data-testid="button-Cancel"]',
+    );
+    this.userNameField = this.page.locator(
+      'xpath=//input[@id="inputUserName"]',
+    );
+    this.yearOfBirthField = this.page.locator(
+      'xpath=//input[@id="inputYearOfBirth"]',
+    );
+    this.genderField = this.page.locator('xpath=//select[@id="selectGender"]');
+    this.maxUserNameLength = 14;
+    this.minUserNameLength = 3;
+  }
+
+  async navigateToAddUserPage() {
+    await this.page.goto(URLS.ADDUSER);
```

Comment:
It will be better to move it to the basePage.ts and rework this method as generic: 
  async gotToPage(string source) {
    await this.page.goto(source);
    }